Explorer/Assets/DCL/SDKComponents/MediaStream/Systems/CreateMediaPlayerSystem.cs

```diff
@@ -44,48 +51,52 @@ protected override void Update(float t)
         [None(typeof(MediaPlayerComponent))]
         private void CreateAudioStream(in Entity entity, ref PBAudioStream sdkComponent)
         {
-            if (!frameTimeBudget.TrySpendBudget()) return;
-
-            MediaPlayerComponent component = CreateMediaPlayerComponent(sdkComponent.Url, sdkComponent.HasVolume, sdkComponent.Volume, autoPlay: sdkComponent.HasPlaying && sdkComponent.Playing);
-            World.Add(entity, component);
+            CreateMediaPlayer(entity, sdkComponent.Url, sdkComponent.HasVolume, sdkComponent.Volume, autoPlay: sdkComponent.HasPlaying && sdkComponent.Playing);
         }
 
         [Query]
         [None(typeof(MediaPlayerComponent))]
         [All(typeof(VideoTextureComponent))]
-        private void CreateVideoPlayer(in Entity entity, ref PBVideoPlayer sdkComponent)
+        private void CreateVideoPlayer(in Entity entity, PBVideoPlayer sdkComponent)
+        {
+            CreateMediaPlayer(entity, sdkComponent.Src, sdkComponent.HasVolume, sdkComponent.Volume, sdkComponent.HasPlaying && sdkComponent.Playing, OnComplete);
+            return;
+
+            void OnComplete(MediaPlayer mediaPlayer) => mediaPlayer.SetPlaybackProperties(sdkComponent);
+        }
+
+        private void CreateMediaPlayer(in Entity entity, string url, bool hasVolume, float volume, bool autoPlay, Action<MediaPlayer> onComplete = null)
         {
             if (!frameTimeBudget.TrySpendBudget()) return;
 
-            MediaPlayerComponent component = CreateMediaPlayerComponent(sdkComponent.Src, sdkComponent.HasVolume, sdkComponent.Volume, autoPlay: sdkComponent.HasPlaying && sdkComponent.Playing);
+            MediaPlayerComponent component = CreateMediaPlayerComponent(url, hasVolume, volume);
 
             if (component.State != VideoState.VsError)
-                component.MediaPlayer.SetPlaybackProperties(sdkComponent);
+            {
+                MediaPlayer mediaPlayer = component.MediaPlayer;
+                mediaPlayer.OpenMediaIfReachableAsync(webRequestController, component.URL, autoPlay, component.Cts.Token, OnComplete).Forget();
```

Comment:
I think you should `SuppressCancellationThrow()` as it's a legit scenario, not an exception